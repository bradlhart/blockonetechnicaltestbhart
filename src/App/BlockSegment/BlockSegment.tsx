import React, { useState } from 'react';

export const BlockSegment: React.FC = ({ blockInfo }) => {
  const [expanded, setExpanded] = useState(false);

  const expandRawData = () => setExpanded(!expanded);

  return (
    <div className="ui segment" onClick={expandRawData}>
      <div className="ui three column divided grid">
        <div className="ten wide column">{blockInfo.id}</div>
        <div className="three wide column">{blockInfo.timestamp}</div>
        <div className="three wide column">{blockInfo.transactions ? blockInfo.transactions.length : null}</div>
      </div>
      {expanded ? (
        <div className="ui segment" style={{ maxHeight: '200px', overflow: 'auto' }}>
          <pre>{JSON.stringify(blockInfo, null, 2)}</pre>
        </div>
      ) : null}
    </div>
  );
};
