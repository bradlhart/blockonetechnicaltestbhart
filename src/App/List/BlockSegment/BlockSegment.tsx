import React, { useState } from 'react';
import { GetBlockResult } from 'eosjs/dist/eosjs-rpc-interfaces';
import './BlockSegment.css';

interface BlockSegmentProps {
  blockInfo: GetBlockResult;
}

export const BlockSegment: React.FC<BlockSegmentProps> = ({ blockInfo }: BlockSegmentProps) => {
  const [expanded, setExpanded] = useState<boolean>(false);

  const expandRawData = () => setExpanded(!expanded);

  return (
    <div className="ui segment" onClick={expandRawData}>
      <div className="ui three column divided grid listContent">
        <div className="seven wide mobile eight wide tablet ten wide computer column">{blockInfo.id}</div>
        <div className="five wide mobile five wide tablet four wide computer column">{blockInfo.timestamp}</div>
        <div className="four wide mobile three wide tablet two wide computer column">{blockInfo.transactions ? blockInfo.transactions.length : null}</div>
      </div>
      {expanded ? (
        <div className="ui segment rawDisplayContainer">
          <pre>{JSON.stringify(blockInfo, null, 2)}</pre>
        </div>
      ) : null}
    </div>
  );
};
