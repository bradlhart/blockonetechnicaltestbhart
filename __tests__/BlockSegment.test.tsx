import React from 'react';
import { shallow, ShallowWrapper } from "enzyme";
import { render, fireEvent, getByTestId} from "@testing-library/react";

import BlockSegment  from '../src/App/List/BlockSegment/BlockSegment';

const GetBlockResultMock = {
    timestamp: "timestamp",
    producer: "producer",
    confirmed: 0,
    previous: "previous",
    transaction_mroot: "transaction_mroot",
    action_mroot: "action_mroot",
    schedule_version: 0,
    producer_signature: "producer_signature",
    id: "id",
    block_num: 0,
    ref_block_prefix: 0,
};

describe('BlockSegment', () => {
    let wrapper: ShallowWrapper<any>;
    const props = {
        blockInfo: GetBlockResultMock
    };

    beforeEach(() => {
        wrapper = shallow(<BlockSegment {...props} />);
    });

    test('renders correctly with default props (without transactions)', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('renders correctly with default props (with transactions)', () => {
        const blockInfo = props.blockInfo.transactions = [{}];
        wrapper.setProps({...props, blockInfo});
        expect(wrapper).toMatchSnapshot();
    });

    test('expands raw data container when clicking on row', () => {
        const props = {
            blockInfo: GetBlockResultMock
        };

        const { container } = render(<BlockSegment {...props} />);
        const rowContainer = getByTestId(container, "rowContainer");
        expect(rowContainer.children.length).toEqual(1);
        fireEvent.click(rowContainer);
        expect(rowContainer.children.length).toEqual(2);
        fireEvent.click(rowContainer);
        expect(rowContainer.children.length).toEqual(1);
    });
});