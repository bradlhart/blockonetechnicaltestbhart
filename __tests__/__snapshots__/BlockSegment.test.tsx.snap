// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockSegment renders correctly with default props (with transactions) 1`] = `
<div
  className="ui segment"
  data-testid="rowContainer"
  onClick={[Function]}
>
  <div
    className="ui three column divided grid listContent"
  >
    <div
      className="seven wide mobile eight wide tablet ten wide computer column"
    />
    <div
      className="five wide mobile five wide tablet four wide computer column"
    />
    <div
      className="four wide mobile three wide tablet two wide computer column"
    >
      0
    </div>
  </div>
</div>
`;

exports[`BlockSegment renders correctly with default props (without transactions) 1`] = `
<div
  className="ui segment"
  data-testid="rowContainer"
  onClick={[Function]}
>
  <div
    className="ui three column divided grid listContent"
  >
    <div
      className="seven wide mobile eight wide tablet ten wide computer column"
    >
      id
    </div>
    <div
      className="five wide mobile five wide tablet four wide computer column"
    >
      timestamp
    </div>
    <div
      className="four wide mobile three wide tablet two wide computer column"
    >
      0
    </div>
  </div>
</div>
`;
